File: src/main/java/xbony2/rubymod/RubyGem.java
package xbony2.rubymod;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;

public class RubyGem extends Item{

	public RubyGem(){
		super();
		this.setCreativeTab(CreativeTabs.tabMaterials);
		this.setTextureName("bonyrubymod:rubygem");
		this.setUnlocalizedName("gemRuby");
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public void registerIcons(IIconRegister register){
		this.itemIcon = register.registerIcon("bonyrubymod" + ":" + "rubygem");
	}
}

File: src/main/java/xbony2/rubymod/RubyGem.rb
# RubyGem.rb
require 'java'

class RubyGem
  extend net.minecraft.item.Item
  def RubyGem()
    super
    setCreativeTab(CreativeTabs.tabMaterials)
    setTextureName("bonyrubymod:rubygem")
    setUnlocalizedName("gemRuby")
  end
  
  def registerIcons(register)
    itemIcon = register.registerIcon("bonyrubymod" + ":" + "rubygem")
  end
end